crates/client-api/src/routes/database.rs

```diff
@@ -772,46 +592,62 @@ pub async fn set_name<S: ControlStateDelegate>(
         // TODO: better error code handling
         .map_err(log_and_500)?;
 
-    Ok(axum::Json(response))
-}
+    let code = match response {
+        name::InsertDomainResult::Success { .. } => StatusCode::OK,
+        name::InsertDomainResult::TldNotRegistered { .. } => StatusCode::BAD_REQUEST,
+        name::InsertDomainResult::PermissionDenied { .. } => StatusCode::UNAUTHORIZED,
+        name::InsertDomainResult::OtherError(_) => StatusCode::INTERNAL_SERVER_ERROR,
+    };
 
-/// This API call is just designed to allow clients to determine whether or not they can
-/// establish a connection to SpacetimeDB. This API call doesn't actually do anything.
-pub async fn ping<S>(State(_ctx): State<S>, _auth: SpacetimeAuthHeader) -> axum::response::Result<impl IntoResponse> {
-    Ok(())
+    Ok((code, axum::Json(response)))
 }
 
-pub fn control_routes<S>(ctx: S) -> axum::Router<S>
+pub fn routes<S>(ctx: S, proxy: impl ControlProxy<S>) -> axum::Router<S>
 where
     S: NodeDelegate + ControlStateDelegate + Clone + 'static,
 {
-    use axum::routing::{get, post};
+    use axum::routing::{delete, get, post, put};
+    let publish = publish::<S>.layer(DefaultBodyLimit::disable());
+    let db_router = axum::Router::<S>::new()
+        .route("/", put(publish.clone()))
+        .route("/", get(db_info::<S>))
+        .route("/", delete(delete_database::<S>))
+        .route("/names", get(get_name::<S>))
+        .route("/names", post(add_name::<S>))
```

Comment:
You cannot set multiple names based on my reading of `add_name`. This should be renamed `set-name` or `add-name`. I believe we should remove the functionality to add multiple names for now tbh.

Reply:
We have no way to remove names. The old functionality was called "set-name" but in reality it just added a new name without removing the old one, by my reading.